python/ray/train/v2/_internal/execution/worker_group/worker_group.py

```diff
@@ -207,6 +209,30 @@ def _start(
 
         assert self.has_started(), "Worker group failed to start."
 
+    def _check_cluster_resources_and_raise_if_insufficient(
+        self, resources_per_worker: Dict[str, float], num_workers: int
+    ):
+        """Check if the cluster has enough resources before waiting for placement group.
+
+        Args:
+            resources_per_worker: The resources per worker.
+            num_workers: The number of workers.
+        """
+        max_cluster_resources = ray_state.get_max_resources_from_cluster_config()
+        if max_cluster_resources and isinstance(max_cluster_resources, dict):
+            for (
+                resource_name,
+                required_amount,
+            ) in resources_per_worker.items():
+                total_required_amount = required_amount * num_workers
+                available_amount = max_cluster_resources.get(resource_name, 0)
```

Comment:
there's an edge case where you can't create the placement group onto the nodes even though the total is satisfied:
```
4 nodes with 8 CPUs --> 32 CPUs
Each worker needs 5 CPUs
Each node can only fit 1 worker --> 4 total workers
Even though 32 > 30 CPUs for 6 workers
```

This is not high priority since most workloads just require 1 GPU per worker, which doesn't run into this issue.

Reply:
+1 can we add a TODO for this? Though it might be more practical to wait for the validation to be completed at the placement group level instead.